pkg/controller/reconciler/kubeprometheusstack_reconciler.go

```diff
@@ -0,0 +1,67 @@
+package reconciler
+
+import (
+	"fmt"
+
+	"github.com/oslokommune/okctl/pkg/config/state"
+
+	"github.com/oslokommune/okctl/pkg/client"
+	"github.com/oslokommune/okctl/pkg/controller/resourcetree"
+)
+
+// KubePromStackState contains runtime data needed in Reconcile()
+type KubePromStackState struct {
+	HostedZone *state.HostedZone
+
+	UserPoolID string
+	AuthDomain string
+}
+
+type kubePrometheusStackReconciler struct {
+	commonMetadata *resourcetree.CommonMetadata
+
+	client client.KubePromStackService
+}
+
+// NodeType returns the resource node type
+func (z *kubePrometheusStackReconciler) NodeType() resourcetree.ResourceNodeType {
+	return resourcetree.ResourceNodeTypeKubePromStack
+}
+
+// SetCommonMetadata saves common metadata for use in Reconcile()
+func (z *kubePrometheusStackReconciler) SetCommonMetadata(metadata *resourcetree.CommonMetadata) {
+	z.commonMetadata = metadata
+}
+
+// Reconcile knows how to do what is necessary to ensure the desired state is achieved
+func (z *kubePrometheusStackReconciler) Reconcile(node *resourcetree.ResourceNode) (*ReconcilationResult, error) {
```

Comment:
Funny. I used 'z' for the hosted Zone reconciler, which was the first reconciler I made. By using the hosted zone reconciler as a template for the others, the 'z' stuck around. And now you're doing the same :D